roslibrust/examples/native_ros1.rs

```diff
@@ -0,0 +1,88 @@
+/**
+ * Testing ground for hitting on rosmaster directly
+ */
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+    simple_logger::SimpleLogger::new()
+        .with_level(log::LevelFilter::Debug)
+        .without_timestamps() // required for running wsl2
+        .init()
+        .unwrap();
+
+    // XMLrpc crate looks archived and doesn't support async, but this worked out of the box
+    // let request = xmlrpc::Request::new("getTopicTypes").arg("my_id");
+    // let result = request.call_url("http://localhost:11311");
+    // println!("Result: {result:?}");
+
+    // rosrust uses a crate xml_rpc, but again no async
+    // Skipped testing it out
+
+    // 3rd try is the charm, serde_xmlrpc looks like it'll do the job, but we need to bring our
+    // own http client. Shouldn't be an issue just use hyper (nevermind use reqwest)
+    let client = roslibrust::MasterClient::new(
+        "http://localhost:11311",
+        "http://localhost:11312",
```

Comment:
From an API perspective it probably makes sense to use the new type pattern here since it will be easy to mix these up and isn't clear from looking at the invocation what each URI is for.

Reply:
Yeah, frankly I'm still debating if I want this "raw client" to be a public API at all. Would love your thoughts on that.

I've been using this example as a cheap version of test driven development before I setup actual testing.

The ROS idiomatic action is for your node to automatically decide on its own what the second address is (the location your node is hosting its own server), but that really isn't very "Rust Like". Probably this all gets hidden under a client options builder pattern?

Reply:
I believe it just binds to port 0 and allows the kernel to decide, so I think it's definitely something that the user shouldn't have to know about. I'm not sure if it'll be necessary to parse the ROS_MASTER_URI variable in order to get the host or if we can just use localhost (probably fine for initial implementation).

Reply:
Its actually worse than that:
http://wiki.ros.org/ROS/NetworkSetup

To "do it right" we actually have to look at two different optional Env vars: ROS_IP and/or ROS_HOSTNAME I'm a little unclear where specifically each should be used, but I think we should host our server at "ROS_IP:0" if it is set or at "ROS_HOSTNAME:0" if ROS_IP is not set.